cpp/src/randomforest/randomforest.cuh

```diff
@@ -184,15 +183,13 @@ class RandomForest {
     // Select n_sampled_rows (with replacement) numbers from [0, n_rows) per tree.
     // selected_rows: randomly generated IDs for bootstrapped samples (w/ replacement); a device
     // ptr.
-    MLCommon::device_buffer<unsigned int>* selected_rows[n_streams];
+    rmm::device_uvector<unsigned int>* selected_rows[n_streams];
```

Comment:
Is this not giving a warning? Looks like it's a conflicting declaration as one of the function parameters is already `unsigned int* selected_rows`

Reply:
Furthermore, using `rmm::device_uvector` as a pointer seems like an anti-pattern